server/status_test.go

```diff
@@ -46,18 +46,37 @@ func startStatusServer() *httptest.Server {
 	return httpServer
 }
 
-// TestStatusLocalStacks verifies that goroutine stack traces are available
-// via the /_status/local/stacks endpoint.
-func TestStatusLocalStacks(t *testing.T) {
+// TestStatus verifies that status endpoints return expected results.
+func TestStatus(t *testing.T) {
 	s := startStatusServer()
 	defer s.Close()
-	body, err := getText(s.URL + statusLocalStacksKey)
-	if err != nil {
-		t.Fatal(err)
+
+	testCases := []struct {
+		keyPrefix string
+		expected  string
+	}{
+		{statusKeyPrefix, "{}"},
+		{statusNodesKeyPrefix, "\"nodes\": null"},
+		// Verify match with at least two goroutine stacks.
+		{statusLocalStacksKey, "(?s)goroutine [0-9]+.*goroutine [0-9]+.*"},
 	}
-	// Verify match with at least two goroutine stacks.
-	if matches, err := regexp.MatchString("(?s)goroutine [0-9]+.*goroutine [0-9]+.*", string(body)); !matches || err != nil {
-		t.Errorf("expected match: %t; err nil: %s", matches, err)
+
+	for _, spec := range testCases {
+		contentTypes := []string{"application/json", "application/x-protobuf", "text/yaml"}
+		for _, contentType := range contentTypes {
+			req, err := http.NewRequest("GET", s.URL+spec.keyPrefix, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+			req.Header.Add("Accept", contentType)
+			body, err := sendAdminRequest(req)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if matches, err := regexp.MatchString(spec.expected, string(body)); !matches || err != nil {
```

Comment:
I think you can use `regexp.Match` instead of `regexp.MatchString` here and avoid the conversion of `body` to a `string`.